core/node/provider.go

```diff
@@ -159,32 +155,65 @@ func OfflineProviders() fx.Option {
 	return fx.Provide(provider.NewNoopProvider)
 }
 
-func newProvidingStrategy(onlyPinned, onlyRoots bool) interface{} {
+func mfsProvider(mfsRoot *mfs.Root, fetcher fetcher.Factory) provider.KeyChanFunc {
+	return func(ctx context.Context) (<-chan cid.Cid, error) {
+		rootNode, err := mfsRoot.GetDirectory().GetNode()
+		if err != nil {
+			return nil, fmt.Errorf("error loading mfs root, cannot provide MFS: %w", err)
+		}
+
+		kcf := provider.NewDAGProvider(rootNode.Cid(), fetcher)
+		return kcf(ctx)
+	}
+
+}
+
+func mfsRootProvider(mfsRoot *mfs.Root) provider.KeyChanFunc {
+	return func(ctx context.Context) (<-chan cid.Cid, error) {
+		rootNode, err := mfsRoot.GetDirectory().GetNode()
+		if err != nil {
+			return nil, fmt.Errorf("error loading mfs root, cannot provide MFS: %w", err)
```

Comment:
Is there a reguluar situation in which we don't have an MFS root? I'm guessing it is always set so it is ok to return an error here rather than ignoring?

Reply:
I've seen cases where MFS root could not be loaded due to filesystem corruption  (https://github.com/ipfs/ipfs-desktop/issues/2882#issuecomment-2658038042). 
But that is not a regular case, and we have https://github.com/ipfs/kubo/issues/7183#issuecomment-2658029979 for tracking UX fix for this scenario, to allow users to self-fix.